Link to 0.10.3 tag instead of `master` branch

berfarah opened the pull request:
#### Purpose
Link to latest tagged release so people can read the latest docs for an available release.

#### Changes
Updated links

#### Caveats
These links have to be updated with every release

#### Related GitHub issues
https://github.com/rails-api/active_model_serializers/issues/1988

#### Additional helpful information
It may be less confusing for new users if the docs link them to the current stable release.

bf4 commented on README.md:
```diff
@@ -84,8 +84,8 @@ If you'd like to chat, we have a [community slack](http://amserializers.herokuap
 ## Documentation
-- [0.10 (master) Documentation](https://github.com/rails-api/active_model_serializers/tree/master)
-  - [![API Docs](http://img.shields.io/badge/yard-docs-blue.svg)](http://www.rubydoc.info/gems/active_model_serializers/0.10.2)
+- [0.10 (master) Documentation](https://github.com/rails-api/active_model_serializers/commit/cd09e890064d2611f2977975163224e12e119568)
```
should be `https://github.com/rails-api/active_model_serializers/tree/v0.10.3`  but really we need a note like:

```
**If you're viewing this at https://github.com/rails-api/active_model_serializers,
 you're reading the documentation for the master branch.
 View documentation for the latest release.... etc
```

berfarah replied:
Sounds good!

bf4 commented on README.md:
```diff
@@ -84,8 +84,8 @@ If you'd like to chat, we have a [community slack](http://amserializers.herokuap
 Thanks!
 
 ## Documentation
-- [0.10 (master) Documentation](https://github.com/rails-api/active_model_serializers/tree/master)
```
it does say `0.10 (master` `¯\_(ツ)_/¯`

berfarah replied:
Gooooooooood point. Would you like me to create a link for `[0.10.3]` instead and keep master there? Or shall I just rename it?

bf4 commented:
@berfarah ping me when you've finished the updates.  Let's keep the link to master and add a link to the most recent released version. That would probably be most useful.  And we should highlight the release.  I think the master was there because there had been some long gaps between releases.

berfarah commented:
@bf4 I've updated the commit to highlight the latest release, keep master and added a note about which version of the docs people would be reading. Let me know what you think :)

bf4 commented on README.md:
```diff
@@ -84,8 +84,14 @@ If you'd like to chat, we have a [community slack](http://amserializers.herokuap
 
 ## Documentation
+
+If you're reading this at github.com/rails-api/active_model_serializers you are
```
`https://github etc`
